### PR TITLE
Add a HomogeneousViewport class that works like MixedViewport but handles only children that have all the same height.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

File: sky/packages/sky/lib/widgets/homogeneous_viewport.dart [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    try {
+      double mainAxisExtent = direction == BlockDirection.vertical ? constraints.maxHeight : constraints.maxWidth;
+      int firstIndex;
+      if (startOffset <= 0.0) {
+        firstIndex = 0;
+      } else {
+        firstIndex = startOffset ~/ itemExtent;
+      }
+      int count;
+      if (mainAxisExtent < double.INFINITY) {
```
Comment (abarth) [inline sky/packages/sky/lib/widgets/homogeneous_viewport.dart]: This if-statement looks flipped...

Comment (Hixie) [inline sky/packages/sky/lib/widgets/homogeneous_viewport.dart]: Negative startOffset means that it's scrolled down so the first item is lower than the top, no? If I scroll the list to half-way down the list, is startOffset positive or negative?

File: sky/packages/sky/lib/widgets/homogeneous_viewport.dart [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        count = itemCount - firstIndex;
+      } else {
+        count = (mainAxisExtent / itemExtent).ceil() + 1;
+      }
+      _children = builder(firstIndex, count);
```
Comment (Hixie) [inline sky/packages/sky/lib/widgets/homogeneous_viewport.dart]: This needs to be:
`dart
List<Widget> newChildren = builder(firstIndex, count);
syncChildren(newChildren, _children);
_children = newChildren;
`

File: sky/packages/sky/lib/widgets/homogeneous_viewport.dart (hunk 1 above)
Comment (abarth) [inline sky/packages/sky/lib/widgets/homogeneous_viewport.dart]: positive

I more meant that in the "else" case below you use mainAxisExtent but in that case its probably double.INFINITY.

Comment (Hixie) [inline sky/packages/sky/lib/widgets/homogeneous_viewport.dart]: Ah yeah, the second one was flipped. Oops.

**Comment by Hixie** [comment]
(this PR is defunct for now)

**Comment by Hixie** [comment]
Ready for review. @abarth

**Pull request merged**